pkg/metrics/client_go_adapter.go

```diff
@@ -152,20 +84,6 @@ func registerClientMetrics() {
 	})
 }
 
-// registerReflectorMetrics sets up reflector (reconcile) loop metrics.
```

Comment:
Why are we removing these ones? Will we still export the same?
is not it a breaking change?

Reply:
These metrics have not been exported for a long time since cliengo has removed them, so the deletion of these code won't cause break changes on current metrics behavior.

Reply:
Kubebuilder uses controller runtime, and the metrics exported by default are doc in: https://book.kubebuilder.io/reference/metrics-reference.html.

@varshaprasad96 wdyt?

Reply:
It seems that these reflector metrics are not doced as expected.

Reply:
HI @zqzten, 

If we check it with client-go 0.24 and controller-runtime 0.12.1, we will no longer see these metrics exported?

Reply:
@camilamacedo86 For the first question, yes, they are no longer exported so we could no longer see these metrics. For the last question, yes, users can implement them as they want, but it's meaningless since cliengo is no longer using this metrics provider, the implementation would not be called.

Reply:
There might be some misuderstanding, these metrics are no longer exported since a very early version (k8s 1.14), so this PR won't bring any changes to current exporting metrics.

Speaking of the NOTE, there might be an alternative that we just not remove these code and add a NOTE to say they are no longer exported, which do you prefer?

Reply:
@zqzten,

You are saying:

> There might be some misuderstanding, these metrics are no longer exported since a very early version (k8s 1.14), so this PR won't bring any changes to current exporting metrics

So, how we can check these metrics exported in https://github.com/kubernetes-sigs/controller-runtime/pull/1901#discussion_r879200489?

Reply:
just to make things clear, I could view the metrics in https://book.kubebuilder.io/reference/metrics-reference.html, but not the those which removed in https://github.com/kubernetes/kubernetes/pull/74636

Reply:
HI @zqzten,

Sorry, after re-checking it better, I saw that the metrics we are removing here are not the same as those exported. My apologies for do not check this before. 

I think it is all fine and make more sense remove the unuseful code them keep it 👍